### PR TITLE
[GDScript/en] Add GDScript

### PR DESCRIPTION
**Pull request by Wichamir**
Made a tutorial for Godot's GDScript. Related to #4242
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: gdscript.html.markdown [hunk 1]
```diff
@@ -0,0 +1,316 @@
+---
+language: GDScript
+contributors:
+    - ["Wichamir", "https://github.com/Wichamir/"]
+filename: learn_gdscript.gd
```
Comment (Menelion) [inline gdscript.html.markdown:5]:
```suggestion
filename: learngdscript.gd
```

File: gdscript.html.markdown [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+---
+
+GDScript is a dynamically typed scripting language made specifically for
+free and open source game engine Godot. GDScript's syntax is similar to
+Python's. It's main advantages are ease of use and tight integration with
```
Comment (Menelion) [inline gdscript.html.markdown:10]:
```suggestion
Python's. Its main advantages are ease of use and tight integration with
```

File: gdscript.html.markdown [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+      continue # 6 won't get printed because of continue statement
+    prints("x is equal to:", x)
+    if x == 7:
+      break # loop will break on 7, so 8, 9 and 10 won't get printed
+
+  match x:
+    1:
+      print("Match is similar to switch.")
+    2:
+      print("However you don't neeed to put cases before each value.")
```
Comment (Menelion) [inline gdscript.html.markdown:138]:
```suggestion
      print("However you don't need to put cases before each value.")
```

File: gdscript.html.markdown [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    _:
+      print("Underscore is a default case.")
+  
+  # ternary operator (one line if-else statement)
+  prints("x is", "positive" if x >= 0 else "negative")
+
+# Casting
+func casting_examples():
+  var i = 42
+  var f = float(42) # cast using variables contructor
```
Comment (Menelion) [inline gdscript.html.markdown:154]:
```suggestion
  var f = float(42) # cast using variables constructor
```

File: gdscript.html.markdown [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+  var b = i as bool # or using "as" keyword
+
+# Override functions
+# By a convention built-in overridable functions start with an underscore,
+# but in practice you can override virtually any function.
+
+# _init gets called when object gets initialized
```
Comment (Menelion) [inline gdscript.html.markdown:161]:
```suggestion
# _init is called when object gets initialized
```

File: gdscript.html.markdown [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+# This is the object's contructor.
```
Comment (Menelion) [inline gdscript.html.markdown:162]:
```suggestion
# This is the object's constructor.
```

File: gdscript.html.markdown [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+func _init():
+  # Initialize object's internal stuff here.
+  pass
+
+# _ready gets called when when script's node and
```
Comment (Menelion) [inline gdscript.html.markdown:167]:
```suggestion
# _ready gets called when script's node and
```

File: gdscript.html.markdown [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+func _ready():
+  pass
+
+# _process gets called on every frame.
+func _process(delta):
+  # The delta argument passed to this function is a number of seconds,
+  # which passed between the last frame and the current one.
+  print("Delta time equals: ", delta)
+
+# _physics_process gets called every physics frame.
```
Comment (Menelion) [inline gdscript.html.markdown:178]:
```suggestion
# _physics_process gets called on every physics frame.
```

File: gdscript.html.markdown [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+```nim
+extends Node
+
+var timer
+
+# You can get references to other nodes in _ready.
+func _ready() -> void:
+  # NodePath is useful for accessing nodes.
+  # Create NodePath by passing String to its constructor:
+  var path1 = NodePath("path/to/someting")
```
Comment (Menelion) [inline gdscript.html.markdown:218]:
```suggestion
  var path1 = NodePath("path/to/something")
```

File: gdscript.html.markdown [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+onready var sprite = $Sprite as Sprite
+
+# You can export NodePath, so you can assign it within the inspector.
+export var nodepath = @""
+onready var reference = get_node(nodepath) as Node
+```
+
+## Signals
+
+Signal system is Godot's implementation of observer programming
```
Comment (Menelion) [inline gdscript.html.markdown:250]:
```suggestion
Signal system is Godot's implementation of the observer programming
```

File: gdscript.html.markdown [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+var x: int # define typed variable
+var y: float = 4.2
+var z := 1.0 # infer type based on default value using := operator
+
+onready var node_ref_typed := $Child as Node
+
+export var speed := 50.0
+
+const CONSTANT := "Typed contant."
```
Comment (Menelion) [inline gdscript.html.markdown:290]:
```suggestion
const CONSTANT := "Typed constant."
```

**Review by Menelion — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please review my suggestions and I'll merge.

**Comment by Wichamir** [comment]
Ok, I read through the whole thing again. It turned out I somehow mismatched types in "Accessing other nodes in the scene tree" section and did some other small mistakes there, so I fixed those. I also reviewed your suggestions. Thanks for checking for typos.

**Review by Menelion — APPROVED** [review]
Thank you!

**Pull request merged**